gengokit/gengokit.go

```diff
@@ -160,134 +266,62 @@ func (g *generator) updateClientMethods(clientHandler io.Reader, svcFuncs []stri
-	clientHandlerFile := getFileByName(packageName+clientHandlerFilePath, previousFiles)
-
-	for _, templateFilePath := range templateFileAssets.AssetNames() {
-		if filepath.Ext(templateFilePath) != ".gotemplate" {
-			log.WithField("Template file", templateFilePath).Debug("Skipping rendering non-buildable partial template")
-			continue
-		}
-
-		var generatedFilePath string
-		var generatedCode *bytes.Buffer
-		var err error
-
-		if templateFilePath == "NAME"+serverHandlerFilePath+"template" && serverHandlerFile != nil {
-			// If there's an existing service file, update its contents
-
-			generatedCode, err = g.updateServerMethods(serverHandlerFile, serviceFunctions)
-
-			if err != nil {
-				return nil, errors.Wrap(err, "could not modifiy service handler file")
-			}
-
-		} else if templateFilePath == "NAME"+clientHandlerFilePath+"template" && clientHandlerFile != nil {
-			// If there's an existing client_handler file, update its contents
-
-			generatedCode, err = g.updateClientMethods(clientHandlerFile, serviceFunctions)
-
-			if err != nil {
-				return nil, errors.Wrap(err, "could not modifiy client handler file")
-			}
-
-		} else {
+	svcFuncs = append(svcFuncs, "NewService")
 
-			generatedCode, err = applyTemplateFromPath(templateFilePath, g.templateExec)
-			if err != nil {
-				return nil, errors.Wrap(err, "could not render template")
-			}
-		}
-
-		generatedFilePath = templateFilePath
-		// Change file path from .gotemplate to .go
-		generatedFilePath = strings.TrimSuffix(generatedFilePath, "template")
-
-		// Turn code buffer into string and format it
-		formattedCode := formatCode(generatedCode.Bytes())
-
-		var resp truss.SimpleFile
-
-		// Switch "NAME" in path with packageName.
-		//i.e. for packageName = addsvc; /NAME-service/NAME-server -> /addsvc-service/addsvc-server
-		resp.Path = strings.Replace(generatedFilePath, "NAME", packageName, -1)
-
-		resp.Write(formattedCode)
-
-		codeGenFiles = append(codeGenFiles, &resp)
-	}
-
-	return codeGenFiles, nil
+	return svcFuncs
 }
 
-// applyTemplateForMissingMeths accepts a funcIndex which represents functions
-// already present in a gofile, applyTemplateForMissingMeths compares this map
-// to the functions defined in the service and renders the template with the
-// path of templPath, and appends this to passed code
-func (g *generator) applyTemplateForMissingMeths(templPath string, funcIndex map[string]bool, code *bytes.Buffer) error {
+// trimTemplateExecutorServiceFuncs removes functions in funcsInFile from the
+// passed templateExecutor and returns a pointer to that templateExecutor
+func trimTemplateExecutorServiceFuncs(te templateExecutor, funcsInFile map[string]bool) (*templateExecutor, error) {
 	var methodsToTemplate []*deftree.ServiceMethod
-	for _, meth := range g.templateExec.Service.Methods {
+
+	for _, meth := range te.Service.Methods {
 		methName := meth.GetName()
-		if funcIndex[methName] == false {
+
+		if funcsInFile[methName] == false {
```

Comment:
I'm suspicious about `funcsInFile`. If the key does not exist, then `false` will be returned since that's the zero value of a `bool`. This seems more reliable:
```
funcsInFile map[string]struct{}
...
if _, inFile := funcsInFile[methName]; inFile {
    continue
}
methodsToTemplate = append(methodsToTemplate, meth)
log.WithField("Method", methName).Info("Rendering template for method")
```